lib/charms/opensearch/v0/opensearch_relation_provider.py

```diff
@@ -120,22 +164,26 @@ def _on_index_requested(self, event: IndexRequestedEvent) -> None:
         """
         if not self.unit.is_leader():
             return
-        if not self.opensearch.is_node_up():
+        if not self.opensearch.is_node_up() or not event.index:
```

Comment:
If it's not provided in the databag for whatever reason, then it evaluates to false. Maybe it's a bit too defensive, I'll try removing it and running tests

Reply:
Yes - I think my point was about having some predictability and confidence in the code and knowing exactly what might cause what.
We need to have validation at a single point of entry so that we assume everywhere else in the code the inputs are right (later PR perhaps?).